### ISSUE TITLE
Duplicate owner reference in packageserver service

### ISSUE THREAD

**Issue by bryanl** [operator-framework/operator-lifecycle-manager#1237]
## Bug Report

**What did you do?**

Installed OLM using kubectl

**What did you expect to see?**

The service created should have a single owner reference to the CSV. 

**What did you see instead? Under which circumstances?**

The service has multiple owner references to the same CSV.

```yaml
---
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2020-01-15T13:49:15Z"
  name: v1-packages-operators-coreos-com
  namespace: olm
  ownerReferences:
  - apiVersion: operators.coreos.com/v1alpha1
    blockOwnerDeletion: false
    controller: false
    kind: ClusterServiceVersion
    name: packageserver
    uid: e2bdc207-8f58-4ef8-afe4-212704dbe060
  - apiVersion: operators.coreos.com/v1alpha1
    blockOwnerDeletion: false
    controller: false
    kind: ClusterServiceVersion
    name: packageserver
    uid: e2bdc207-8f58-4ef8-afe4-212704dbe060
  resourceVersion: "201176"
  selfLink: /api/v1/namespaces/olm/services/v1-packages-operators-coreos-com
  uid: 7a1f2c82-e2ae-4adb-a10d-dee40c8a76ae
spec:
  clusterIP: 10.96.68.225
  ports:
  - port: 443
    protocol: TCP
    targetPort: 5443
  selector:
    app: packageserver
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```

**Environment**
* operator-lifecycle-manager version:

https://github.com/operator-framework/operator-lifecycle-manager/tree/976f2b981c326ee51748045f6df20317d6d95bd5

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:20:10Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:12:17Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

Minikube

**Comment by exdx** [comment]
Hey @bryanl, thanks for filing this bug. We think it stems from the way we set owner references without checking the existence of previous references. Will take a look at this one.

**Comment by exdx** [comment]
Looking into this a bit, I see we do check for existing owner references when assigning a new owner to an object, and return early instead of assigning the same reference again. https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/lib/ownerutil/util.go#L158

That being said this bug shows that the duplicate owernerref is being set again. This suggests it could be a race condition. The call that sets the ownerref for the packageserver service is here https://github.com/operator-framework/operator-lifecycle-manager/blob/366ed269986257e5bd55f29bd4e167c7cf7182f8/pkg/controller/operators/olm/apiservices.go#L366. This runs inside of OLM operator and is triggered every time a sync is being done. 

One thing I noticed is that the function that actually sets the ownerref has the following signature. 
```
func AddNonBlockingOwner(object metav1.Object, owner Owner)
``` 
It takes copies to objects instead of pointers to the objects themselves. Therefore in the case where one sync runs, and calls this function, it sees no references and appends them, and if another sync is running and calls this function with the same object, it will not see the updated references yet and set it again. Then as the sync updates the object on the api server it sets duplicate references. I played around with refactoring the signature to take in pointers to the object and owner args instead, but this causes issues like `Cannot use type *Service as type *metav1.Object` as the pointer type fulfills a different interface. 

All that being said if this was on on minikube then the sync should run really fast, and assuming no errors were present I'm a little doubtful its a race. Any thoughts @ecordell @njhale?
